### PR TITLE
panel attachment mime-type error: TypeError

### PR DESCRIPTION
**Pull request by pawelmarkowski**
can't use a string pattern on a bytes-like object

Title changed from TEST to . Null title on attachment upload.

above 2 problems solved

### CODE REVIEW

**Comment by adrianboyko** [comment]
I'm running Django 1.8.5  with Python 3 and ran into this problem on every attempt to attach a second file to a ticket (interestingly, the _first_ attachment _always_ works). I applied Pawel's commit to a branch off the 0.1.17 tag and it fixed the problem.

**Comment by flinz** [comment]
Nice catch - and it seems this is not covered by our tests. I will add this to the Testing project.

File: helpdesk/views/staff.py [hunk 1]
```diff
@@ -457,7 +457,7 @@ def update_ticket(request, ticket_id, public=False):
     if request.FILES:
         import mimetypes, os
         for file in request.FILES.getlist('attachment'):
-            filename = file.name.encode('ascii', 'ignore')
+            filename = file.name
```
Comment (flinz) [inline helpdesk/views/staff.py]: Are you sure this change does not break things in python 2.7 when using non-ascii characters in filenames?
The ``'.encode('ascii', ignore')`` effectively removes non-ascii characters from unicode.

**Comment by flinz** [comment]
Upon a little research, it seems that In fact this ecode was introduced as a "fix" a while ago.
See https://github.com/django-helpdesk/django-helpdesk/issues/162

Simply removing it will not solve this issue. I strongly suggest we add a test that covers attachments, also with non-ascii filenames, then propose a solution that passes these tests.

File: helpdesk/views/staff.py (hunk 1 above)
Comment (gwasser) [inline helpdesk/views/staff.py]: django.utils.six will be required here. Py2/3 handle strings very differently.

Comment (pawelmarkowski) [inline helpdesk/views/staff.py]: It is always a good idea to check. Can not we solve this problem without django.utils.six :)? I think we can :) just check this code on py2.7 it should work even there.

**Comment by reduxionist** [comment]
Closing this PR since, effectively, its contents were merged through #463 and #464.

**Pull request closed without merging**